### PR TITLE
Automatically configure SSH key for user

### PR DESCRIPTION
**Pull request by dabbott**
This needs some testing still, but the idea is:

Can we make a fully automated flow, so that once somebody has a GitHub account, we can set up push access? (everything is opt-in)

This means, if a push fails due to invalid credentials:
- we prompt them to ask if we should set up SSH access
- if they don't have an .ssh key, we create one
- we upload their public key to their github account (which requires `write:public_key`)
- we verify the identity of github.com using a compiled-in SSH fingerprint

I couldn't find a GraphQL mutation for adding a public_key, so I use the API v3 here... but maybe I just missed it?

I thought it was going to be hacky but it's actually not too bad IMO. There's still a bug after github repo creation where it doesn't move to the next part of the flow automatically

### CODE REVIEW

File: studio/LonaStudio/VersionControl/Git.swift [hunk 1]
```diff
@@ -79,81 +101,97 @@ public enum Git {
...
         return run(arguments: ["remote", "add", name, url.absoluteString])
     }
 
-    public func push(repository: String, refspec: String) -> Result<String, NSError> {
-        return run(arguments: ["push", repository, refspec])
+    public func push(repository: String, refspec: String) -> Result<String, GitError> {
+        return run(arguments: ["push", repository, refspec]).mapError({ error in
+            if error.code == 128,
+                let errorString = error.errorOutput.utf8String(),
+                (errorString.contains("git@github.com: Permission denied (publickey)") || errorString.contains("Host key verification failed")) {
```
Comment (mathieudutour) [inline studio/LonaStudio/VersionControl/Git.swift]: in dugite (what GitHub is using), they just check that the code is !== 0 and that's their regex: https://github.com/desktop/dugite/blob/e447c5ef95b2cfd13e98a4d85f89e7906f82dbdc/lib/errors.ts#L60-L63

**Comment by mathieudutour** [comment]
That's really cool 💯

File: studio/LonaStudio/Utils/RESTClient.swift [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+}
+
+extension RESTClient {
+    static var githubV3: RESTClient = {
+        let client = RESTClient(baseURL: URL(string: "https://api.github.com")!)
+
+        client.configureRequest = { request in
+            var request = request
+
+            if let githubToken = Account.shared.cachedMe?.githubAccessToken {
```
Comment (mathieudutour) [inline studio/LonaStudio/Utils/RESTClient.swift]:
```suggestion
            let group = DispatchGroup()
            group.enter()

            Account.shared.me().finalResult({_ in
                group.leave()
            })

            // wait for the Lona account to be fetched
            group.wait()

            if let githubToken = Account.shared.cachedMe?.githubAccessToken {
```

File: studio/LonaStudio/VersionControl/Git.swift (hunk 1 above)
Comment (dabbott) [reply studio/LonaStudio/VersionControl/Git.swift]: :+1: copying theirs

**Pull request merged**